### ISSUE TITLE
Training the default perceptron tagger with bigger corpus

### ISSUE THREAD

**Issue by alvations** [nltk/nltk#1317]
The default pickle for the `PerceptronTagger` is nice but it's a little too small for realistic usage:

``` python
>>> from nltk import PerceptronTagger
>>> len(PerceptronTagger(load=True).tagdict)
1549
```

Would it be possible to retrain a model on something like the full Penn TreeBank or BNC? Is the model then releasable on NLTK?

**Comment by stevenbird** [comment]
Yes, a trained model is not a derived work, so long as the original data cannot be reconstructed from it.

**Comment by alvations** [comment]
@syllog1sm Is there a reason for the high `freq_thresh = 20` at https://github.com/nltk/nltk/blob/develop/nltk/tag/perceptron.py#L268 ?

**Comment by syllog1sm** [comment]
I tried to set the tag dictionary hyper-parameters such that impact on accuracy is really minimal. I don't remember how much I tested this across domains. The POS tagger blog post did include a cross-domain evaluation at the end, but I think I considered this a test set and only did the experiment after I'd finalised the model.

I think you should be quite careful about setting the tag dictionary thresholds more aggressively. It's easy to make the model very brittle to domain changes with this mechanism. Using the tag dictionary for rarer words types also doesn't help much for efficiency. You're only extending the tag dictionary to a few % of tokens.

**Comment by shanuu23** [comment]
Hi @stevenbird , I’m interested in working on this. Could you confirm that it’s still wanted and suggest a suitable redistributable corpus for retraining the model?
